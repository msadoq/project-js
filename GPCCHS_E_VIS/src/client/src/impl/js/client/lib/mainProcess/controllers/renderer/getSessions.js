import getLogger from 'common/log';
import { server } from '../../ipc';
import { getStore } from '../../../store/mainStore';
import { updateSessions } from '../../../store/actions/sessions';

const logger = getLogger('main:controllers:renderer:getSessions');

export default function (reply, event, payload, queryId) {
  server.requestSessions((err, sessions = []) => {
    if (err) {
      return logger.error(err);
    }

    // update sessions list in store
    getStore().dispatch(updateSessions(sessions));

    // inform renderer process that sessions are up to date
    reply({ event: 'runCallback', queryId });
  });
}
