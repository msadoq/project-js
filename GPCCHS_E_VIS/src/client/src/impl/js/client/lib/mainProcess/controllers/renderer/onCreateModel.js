import _cloneDeep from 'lodash/cloneDeep';
import { dirname } from 'path';
import { getStore } from '../../../store/mainStore';
import { getView } from '../../../store/selectors/views';
import { add } from '../../../store/actions/messages';
import { getPathByFilePicker } from '../../dialog';
import { saveViewAs } from '../../../common/documentManager';
import vivl from '../../../../VIVL/main';
import { getRootDir } from '../../../common/fmd';


const root = getRootDir();

export default function ({ viewId }) {
  const { getState, dispatch } = getStore();
  const { type, configuration, absolutePath } = getView(getState(), viewId);

  const folder = absolutePath ? dirname(absolutePath) : root;
  return getPathByFilePicker(folder, 'model', 'save', (err, path) => {
    if (path) {
      // Create a twin view without formula in entry Points
      const modelConfig = _cloneDeep(configuration);
      const structureType = vivl(type, 'structureType')();
      if (structureType === 'last') {
        modelConfig.entryPoints.forEach((ep) => {
          ep.connectedData.formula = ''; // eslint-disable-line no-param-reassign
        });
      } else if (structureType === 'range') {
        modelConfig.entryPoints.forEach((ep) => {
          ep.connectedDataX.formula = ''; // eslint-disable-line no-param-reassign
          ep.connectedDataY.formula = ''; // eslint-disable-line no-param-reassign
        });
      }

      saveViewAs(modelConfig, type, path, (errSaving) => {
        if (errSaving) {
          dispatch(add(viewId, 'danger', `Model unsaved ${errSaving}`));
        } else {
          dispatch(add(viewId, 'success', 'Model saved'));
        }
      });
    }
  });
}
