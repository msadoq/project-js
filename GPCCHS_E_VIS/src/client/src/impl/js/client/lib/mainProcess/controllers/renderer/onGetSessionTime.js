import getLogger from 'common/log';
import reply from 'common/ipc/reply';
import { server } from '../../ipc';

const logger = getLogger('main:controllers:renderer:onGetSessionTime');

export default function (queryId) {
  server.requestSessionTime(({ err, time }) => {
    if (err) {
      return logger.error(err);
    }

    reply(queryId, time);
  });
}
