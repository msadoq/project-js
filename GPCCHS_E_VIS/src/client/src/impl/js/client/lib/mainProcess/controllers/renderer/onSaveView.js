import { dirname } from 'path';
import { getStore } from '../../../store/mainStore';
import { getView } from '../../../store/selectors/views';
import { setModified, updatePath, updateAbsolutePath, setViewOid } from '../../../store/actions/views';
import { addOnce as addMessage } from '../../../store/actions/messages';
import { getPathByFilePicker } from '../../dialog';
import { saveViewAs } from '../../../common/documentManager';

import { getRootDir, isFmd, getRelativeFmdPath } from '../../../common/fmd';

const getPath = path => (isFmd(path) ? getRelativeFmdPath(path) : path);

const root = getRootDir();
const addViewError = (viewId, msg) => addMessage(viewId, 'danger', msg);

export default function ({ viewId, saveAs }) {
  const { getState, dispatch } = getStore();
  const { type, configuration, absolutePath } = getView(getState(), viewId);

  function oncePath(savingAbsolutePath) {
    saveViewAs(configuration, type, savingAbsolutePath, (err, oid) => {
      if (err) {
        return getStore().dispatch(addViewError(viewId, err));
      }
      if (saveAs !== savingAbsolutePath) {
        // only for 'Save as...' action
        dispatch(updatePath(viewId, getPath(savingAbsolutePath)));
        dispatch(updateAbsolutePath(viewId, savingAbsolutePath));
        if (oid) {
          dispatch(setViewOid(viewId, oid));
        }
      }

      dispatch(setModified(viewId, false));
      dispatch(addMessage(viewId, 'success', 'View saved'));
    });
  }

  if (saveAs) {
    return oncePath(saveAs);
  }

  const folder = absolutePath ? dirname(absolutePath) : root;
  return getPathByFilePicker(folder, 'view', 'save', (err, path) => oncePath(path));
}
