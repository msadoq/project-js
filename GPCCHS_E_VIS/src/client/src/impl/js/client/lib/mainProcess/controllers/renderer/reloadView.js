/**
 * Created by isis on 1/5/17.
 */
