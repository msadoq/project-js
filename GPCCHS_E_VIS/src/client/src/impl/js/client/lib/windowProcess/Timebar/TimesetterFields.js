import moment from 'moment';
import classnames from 'classnames';
import React, { Component } from 'react';
import { Col } from 'react-bootstrap';
import styles from './Timesetter.css';

export default class TimesetterFields extends Component {

  static propTypes = {
    ms: React.PropTypes.number.isRequired,
    value: React.PropTypes.string.isRequired,
    disabled: React.PropTypes.bool.isRequired,
    onChange: React.PropTypes.func.isRequired,
    visuWindow: React.PropTypes.object.isRequired,
    slideWindow: React.PropTypes.object.isRequired,
  }

  state = {
    date: null
  }

  onChangeAction = () => {
    const { value, onChange, visuWindow, slideWindow } = this.props;
    const { date } = this.state;
    if (visuWindow[value]) {
      visuWindow[value] = date.toDate().getTime();
    } else if (value === 'slideLower') {
      slideWindow.lower = date.toDate().getTime();
    } else if (value === 'slideUpper') {
      slideWindow.upper = date.toDate().getTime();
    }
    onChange(
      visuWindow.lower,
      visuWindow.upper,
      visuWindow.current,
      slideWindow.lower,
      slideWindow.upper,
      value
    );
    this.setState({ date: null });
  }

  changeAttr = (a, e) => {
    const el = e.currentTarget;
    const { ms } = this.props;
    let { date } = this.state;
    let attrVal = parseInt(el.value, 10);

    if (!date) {
      date = moment(ms);
      this.setState({ date });
    }

    // UI month 1 equals moment month 0
    if (a === 'months') attrVal -= 1;

    date.set(a, attrVal);

    /*
      Replacing values of the inputs for each date proprety with their moment formatted equivalents.
      Very important because increases can fire other increases values on upper scopes
      ex: + 1 hour can sometimes switch to the next day / month / year
    */
    Object.keys(formats).forEach((key) => {
      this[`${key}El`].value = date.format(formats[key].format);
    });

    this.setState({ date });
  }

  render() {
    const { ms, value, disabled } = this.props;
    const { date } = this.state;
    const arr = date ? dateToArray(date) : dateToArray(moment(ms));

    let submitButton;
    if (!disabled) {
      submitButton = (<input
        className={classnames(styles.submitButton, 'btn', 'btn-sm', 'btn-primary')}
        type="submit"
        disabled={!date}
        onClick={this.onChangeAction}
      />);
    }
    let valueText;
    if (value === 'slideLower') {
      valueText = 'Ext lower cursor';
    } else if (value === 'slideUpper') {
      valueText = 'Ext upper cursor';
    } else {
      valueText = `${value} cursor`;
    }

    return (
      <Col xs={12} className={styles.fieldsContainer}>
        <div className={classnames('text-capitalize', styles.formLabel, { [styles[`formLabel${value}`]]: !disabled })} style={{ width: '100%' }}>
          <b>{valueText}</b>
        </div>
        {
          arr.map((x, i) =>
            <div key={i} className={styles.inputDiv}>
              <input
                type="number"
                ref={(el) => { this[`${x[0]}El`] = el; }}
                key={i}
                className={classnames('form-control', styles.input, styles[`input_${x[0]}`])}
                defaultValue={x[1]}
                disabled={disabled}
                onClick={this.changeAttr.bind(null, x[0])}
                onBlur={this.changeAttr.bind(null, x[0])}
              />
              {['year', 'months'].find(a => a === x[0]) ? <span>-</span> : ''}
              {['hours', 'minutes'].find(a => a === x[0]) ? <span>:</span> : ''}
              {x[0] === 'seconds' ? <span>.</span> : ''}
            </div>
          )
        }
        {submitButton}
      </Col>
    );
  }
}

const formats = {
  year: { format: 'YYYY', parent: null },
  months: { format: 'MM', parent: 'year' },
  date: { format: 'DD', parent: 'months' },
  hours: { format: 'HH', parent: 'date' },
  minutes: { format: 'mm', parent: 'hours' },
  seconds: { format: 'ss', parent: 'minutes' },
  milliseconds: { format: 'SSS', parent: 'seconds' },
};

function dateToArray(m) {
  return [
    ['year', m.format(formats.year.format)],
    ['months', m.format(formats.months.format)],
    ['date', m.format(formats.date.format)],
    ['hours', m.format(formats.hours.format)],
    ['minutes', m.format(formats.minutes.format)],
    ['seconds', m.format(formats.seconds.format)],
    ['milliseconds', m.format(formats.milliseconds.format)]
  ];
}
