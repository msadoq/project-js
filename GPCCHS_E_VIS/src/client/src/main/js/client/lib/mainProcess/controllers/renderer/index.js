// ====================================================================
// HISTORY
// VERSION : 1.1.0 : : : 28/02/2017 : Initial version
// VERSION : 1.1.2 : DM : #3622 : 07/03/2017 : first draft on inspector: retrieve data from rtd on right-click
// VERSION : 1.1.2 : DM : #5822 : 16/03/2017 : resolve a rtd link in the inspector
// VERSION : 1.1.2 : DM : #5828 : 30/03/2017 : Add a F1 button in VIMA to open the docu wiki helper
// VERSION : 1.1.2 : DM : #5822 : 03/04/2017 : merge dev in working branch
// VERSION : 1.1.2 : DM : #5828 : 23/05/2017 : Move common/callback and common/ipc modules in client sub-component
// VERSION : 1.1.2 : DM : #6785 : 12/06/2017 : activate links in views .
// VERSION : 1.1.2 : DM : #5828 : 13/06/2017 : Move common/constants/ in client/ folder
// VERSION : 1.1.2 : DM : #5828 : 14/06/2017 : Cleanup client/ file organization and test helper modules
// VERSION : 1.1.2 : FA : ISIS-FT-2132 : 15/06/2017 : Ask to save before closing view or page
// VERSION : 1.1.2 : DM : #6700 : 16/06/2017 : Add store enhancers helpers code coverage and merge with dev
// VERSION : 1.1.2 : DM : #6700 : 26/06/2017 : Change windowProcess health management to dispatch a Redux action directly.
// VERSION : 1.1.2 : DM : #6785 : 29/06/2017 : Fix opening view link in a new page and read only path for link definition
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : First draft on catalog explorer
// VERSION : 1.1.2 : DM : #6688 : 05/07/2017 : catalog explorer : open, close and browse items
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Remove unused onSavePage ipc renderer to mainProcess controllers
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 06/07/2017 : Replace ipc openPage by askOpenPage redux action
// VERSION : 1.1.2 : FA : #7235 : 18/07/2017 : Clean old code in menuManager/IPC controller
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 18/07/2017 : Fix common/fmd, disable Links feature for now
// VERSION : 1.1.2 : DM : #6700 : 20/07/2017 : Remove obsolete onServerDebug interface, ipcs and model
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about openInspector .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about FMD_GET/FMD_CREATE and RELOAD_SESSIONS/GET_SESSION_TIME
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Clean IPC about documents .
// VERSION : 1.1.2 : FA : ISIS-FT-1964 : 21/07/2017 : Remove serverDebug ipc . .
// VERSION : 1.1.2 : DM : #6700 : 03/08/2017 : Merge branch 'dev' into dbrugne-data
// VERSION : 1.1.2 : FA : #7145 : 04/08/2017 : Clean IPC about opening wiki helper + create a store folder in mainProcess
// END-HISTORY
// ====================================================================

import constants from 'constants';

import handle from 'common/ipc/handle';
import reply from 'common/ipc/reply';

import onReduxCurrentState from './onReduxCurrentState';
import onReduxDispatch from './onReduxDispatch';
import onResolveLink from './onResolveLink';
import onResolveRteLink from './onResolveRteLink';
import onGetRteCatalogs from './onGetRteCatalogs';
import onGetRteItemNames from './onGetRteItemNames';
import onOpenRteItem from './onOpenRteItem';
import onFocusRteItem from './onFocusRteItem';
import onGetRteDomains from './onGetRteDomains';

const controller = {
  [constants.IPC_METHOD_REDUX_CURRENT_STATE]: (...args) => onReduxCurrentState(reply, ...args),
  [constants.IPC_METHOD_REDUX_DISPATCH]: onReduxDispatch,
  [constants.IPC_METHOD_RESOLVE_LINK]: onResolveLink,
  [constants.IPC_METHOD_GET_RTE_DOMAINS]: onGetRteDomains,
  [constants.IPC_METHOD_GET_RTE_CATALOGS]: onGetRteCatalogs,
  [constants.IPC_METHOD_GET_RTE_ITEM_NAMES]: onGetRteItemNames,
  [constants.IPC_METHOD_OPEN_RTE_ITEM]: onOpenRteItem,
  [constants.IPC_METHOD_FOCUS_RTE_ITEM]: onFocusRteItem,
  [constants.IPC_METHOD_RESOLVE_RTE_LINK]: onResolveRteLink,
};

export default (electronEvent, data) => handle(
  controller,
  data,
  response => electronEvent.sender.send('global', response)
);
