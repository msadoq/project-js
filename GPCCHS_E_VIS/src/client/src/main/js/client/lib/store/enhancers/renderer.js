/**
 * Renderer process store enhancer
 *
 * - intercepts dispatched actions => forwards to main process
 * - receives state diff from main process => apply
 */

export default function rendererStoreEnhancer(previousStoreCreator) {
  if (process.type !== 'renderer') {
    throw new Error('rendererStoreEnhancer used in a non-renderer process');
  }

  return function createStore(reducer, initialState) {
    const store = previousStoreCreator(reducer, initialState);

    // TODO : enhance
    const storeDotDispatch = store.dispatch;
    store.dispatch = (action) => {
      console.log('VIMA ACTION on rendererStoreEnhancer');
      storeDotDispatch(action);
    };

    return store;
  };
}
