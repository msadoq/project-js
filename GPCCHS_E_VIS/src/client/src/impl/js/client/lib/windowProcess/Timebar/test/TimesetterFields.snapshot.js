import React from 'react';
import renderer from 'react-test-renderer';
import TimesetterFields from '../TimesetterFields';
import TimebarFixture from '../../../../test/__mocks__/Timebar';

const propsStub = {
  ms: 1480417090581,
  value: 'lower',
  disabled: false,
  onChange: () => null,
  visuWindow: TimebarFixture.visuWindow,
  slideWindow: TimebarFixture.slideWindow,
};

test('TimesetterFields renders correctly', () => {
  const tree = renderer.create(
    <TimesetterFields
      {...propsStub}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
