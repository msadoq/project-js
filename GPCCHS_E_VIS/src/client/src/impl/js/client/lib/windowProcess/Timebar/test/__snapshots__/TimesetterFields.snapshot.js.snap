exports[`test TimesetterFields renders correctly 1`] = `
<div
  className="Timesetter__fieldsContainer___24Fbf col-xs-12">
  <div
    className="text-capitalize Timesetter__formLabel___2eZoL Timesetter__formLabellower___1lP9u"
    style={
      Object {
        "width": "100%",
      }
    }>
    <b>
      lower cursor
    </b>
  </div>
  <div
    className="Timesetter__inputDiv___2_jkV">
    <input
      className="form-control Timesetter__input___1kEVn Timesetter__input_year___3Bqew"
      defaultValue="2016"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      type="number" />
    <span>
      -
    </span>
    
    
  </div>
  <div
    className="Timesetter__inputDiv___2_jkV">
    <input
      className="form-control Timesetter__input___1kEVn Timesetter__input_months___2pmb9"
      defaultValue="11"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      type="number" />
    <span>
      -
    </span>
    
    
  </div>
  <div
    className="Timesetter__inputDiv___2_jkV">
    <input
      className="form-control Timesetter__input___1kEVn Timesetter__input_date___fOWIW"
      defaultValue="29"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      type="number" />
    
    
    
  </div>
  <div
    className="Timesetter__inputDiv___2_jkV">
    <input
      className="form-control Timesetter__input___1kEVn Timesetter__input_hours___vTXPu"
      defaultValue="10"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      type="number" />
    
    <span>
      :
    </span>
    
  </div>
  <div
    className="Timesetter__inputDiv___2_jkV">
    <input
      className="form-control Timesetter__input___1kEVn Timesetter__input_minutes___vvWf-"
      defaultValue="58"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      type="number" />
    
    <span>
      :
    </span>
    
  </div>
  <div
    className="Timesetter__inputDiv___2_jkV">
    <input
      className="form-control Timesetter__input___1kEVn Timesetter__input_seconds___3SasE"
      defaultValue="10"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      type="number" />
    
    
    <span>
      .
    </span>
  </div>
  <div
    className="Timesetter__inputDiv___2_jkV">
    <input
      className="form-control Timesetter__input___1kEVn Timesetter__input_milliseconds___1otkI"
      defaultValue="581"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      type="number" />
    
    
    
  </div>
  <input
    className="Timesetter__submitButton___3mo_t btn btn-sm btn-primary"
    disabled={true}
    onClick={[Function]}
    type="submit" />
</div>
`;
